geps/gep-3539/index.md

```diff
@@ -0,0 +1,200 @@
+
+* Issue: [#3539](https://github.com/kubernetes-sigs/gateway-api/issues/3539)
+* Status: Provisional
+
+## TLDR
+
+Gateway API enables advanced traffic routing and can be used to expose a 
+logical set of pods on a single IP address within a cluster. It can be seen 
+as the next generation ClusterIP providing more flexibility and composability 
+than Service API. This comes at the expense of some additional configuration 
+and manageability burden.
+
+## Goals
+
+* Define Gateway API usage to accomplish ClusterIP Service style behavior
+* Propose DNS layout and record format for ClusterIP Gateway
+* Extend the use of Gateway API to provide NodePort and LoadBalancer Service 
+  type of functionality
+
+## Non-Goals
+
+* Make significant changes to Gateway API
+* Provide path for existing ClusterIP Services in a cluster to migrate to 
+  Gateway API model
+
+## API Changes
+
+* EndpointSelector is recognized as a backend
+* DNS record format for ClusterIP Gateways 
+
+## Introduction
+
+Gateway API provides a generic and composable model for defining L4 and L7 
+routing in Kubernetes. Very simply, it describes how to get traffic into pods. 
+ClusterIP provides similar functionality of an ingress point for routing traffic 
+into pods. As the Gateway API has evolved, there have been discussions around whether 
+it can be a substitute for the increasingly complex and overloaded Service API. This 
+document aims to describe what this could look like in practice, with a focus on 
+ClusterIP and a brief commentary on how the concept design can be extended to 
+accommodate LoadBalancer and NodePort Services.
+
+## Overview
+
+Gateway API can be thought of as decomposing Service API into multiple separable 
+components that allow for definition of the ClusterIP address and listener configuration 
+(Gateway resource), implementation specifics and common configuration (GatewayClass 
+resource), and routing traffic to backends (Route resource).
+
+### Limitations of Service API
+
+Besides what has been discussed in the past about Service API maintainability, evolvability, 
+and complexity concerns, see: https://www.youtube.com/watch?v=Oslwx3hj2Eg, we ran into 
+additional practical concerns that rendered Service API insufficient for the needs at hand.
+
+Service IPs can only be assigned out of the ServiceCIDR range configured for the API server. 
+While Kubernetes 1.31 added a Beta feature that allows for the Extension of Service IP Ranges, 
+there have been use cases where multi-NIC pods (pods with multiple network interfaces) require 
+the flexibility of specifying different ServiceCIDR ranges to be used for ClusterIP services 
+corresponding to the multiple different networks. There are strict traffic splitting and network 
+isolation requirements that demand non-overlapping ServiceCIDR ranges for per-network ClusterIP 
+service groups. Because of the way service definition and IP address allocation are tightly 
```

Comment:
talked offline with @ptrivedi , what is unique across the cluster is the IPAddress, but we can perfectly introduce a `managed-by` label to ServiceCIDR objects so it can be used to identify multiple ranges to be handled by different controllers